File: golang/array_string/merge_alternatif.go
package main

func mergeAlternately(word1 string, word2 string) string {
	result := []byte{}

	i, j := 0, 0

	for i < len(word1) && j < len(word2) {
		result = append(result, word1[i], word2[j])

		i++

		j++
	}

	if i < len(word1) {
		result = append(result, word1[i:]...)
	}

	if j < len(word2) {
		result = append(result, word2[j:]...)
	}

	return string(result)
}

File: java/src/main/java/com/sanedge/interview/array_string/MergeAlternatif.java
package com.sanedge.interview.array_string;

public class MergeAlternatif {

    public String mergeAlternatily(String word1, String word2) {
        StringBuilder result = new StringBuilder();

        int i = 0;
        int j = 0;

        while (i < word1.length() && j < word2.length()) {
            result.append(word1.charAt(i));
            result.append(word2.charAt(j));

            i++;
            j++;
        }

        if (i < word1.length()) {
            result.append(word1.substring(i));
        }

        if (j < word2.length()) {
            result.append(word2.substring(j));
        }

        return result.toString();
    }
}
